adhocracy4/maps/static/a4maps/map_choose_point.js

```diff
@@ -0,0 +1,101 @@
+function createMap (L, baseurl, name) {
+  var basemap = baseurl + '{z}/{x}/{y}.png'
+  var osmAttrib = '&copy; <a href="http://openstreetmap.org/copyright">OpenStreetMap</a> contributors'
+  var baselayer = L.tileLayer(basemap, {maxZoom: 18, attribution: osmAttrib})
+  var map = new L.Map('map_' + name, {scrollWheelZoom: false, zoomControl: false})
+  baselayer.addTo(map)
+  return map
+}
+
+function createMarker ($, L, newlatln, oldlatln, basePolygon, map, name) {
+  var icon = L.icon({
+    iconUrl: '/static/images/map_pin_01_2x.png',
+    shadowUrl: '/static/images/map_shadow_01_2x.png',
+    iconSize: [30, 45],
+    iconAnchor: [15, 45],
+    shadowSize: [40, 54],
+    shadowAnchor: [20, 54],
+    popupAnchor: [0, -45]
+  })
+
+  var marker = L.marker(newlatln, { draggable: true, icon: icon }).addTo(map)
+  marker.on('dragend', function () {
+    var markerInsidePolygon = false
+    basePolygon.getLayers().forEach(function (each) {
+      if (isMarkerInsidePolygon(marker, each)) {
+        markerInsidePolygon = true
+        oldlatln = marker.getLatLng()
+        var shape = marker.toGeoJSON()
+        $('#id_' + name).val(JSON.stringify(shape))
+      }
+    })
+    if (!markerInsidePolygon) {
+      marker.setLatLng(oldlatln)
+    }
+  })
+  return marker
+}
+
+function isMarkerInsidePolygon (marker, poly) {
+  var polyPoints = poly.getLatLngs()
+  var x = marker.getLatLng().lat
+  var y = marker.getLatLng().lng
+
+  var inside = false
+  for (var i = 0, j = polyPoints.length - 1; i < polyPoints.length; j = i++) {
+    var xi = polyPoints[i].lat
+    var yi = polyPoints[i].lng
+    var xj = polyPoints[j].lat
+    var yj = polyPoints[j].lng
+
+    var intersect = ((yi > y) !== (yj > y)) &&
+        (x < (xj - xi) * (y - yi) / (yj - yi) + xi)
+    if (intersect) inside = !inside
+  }
+  return inside
+}
```

Comment:
I guess this is taken from [adhocracy3](https://github.com/liqd/adhocracy3/blob/master/src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Core/Mapping/MappingUtils.ts). Maybe it would be worth to also copy the comment from there.